### ISSUE TITLE
NetworkObject.cs - SerializedRootNetworkBehaviour can never become null again after set

### ISSUE THREAD

**Issue by FREEZX** [FirstGearGames/FishNet#705]
I was having an issue in my scene where a NetworkObject that was part of a prefab was getting its renderers disabled and never re-enabled afterwards. After digging deep and debugging I figured out it was happening because the NetworkObject has its SerializedRootNetworkBehaviour set to a random object that is a child of it, even though my NetworkObject had no parentNob. I'm not sure how this all happened in my case, but I think it would be smart for UpdateNetworkBehaviours to clear out this value if it should be null.

**General**
Unity version: 2022.2.29f1
Fish-Networking version: 4.3.1

**Comment by FirstGearGames** [comment]
I looked into this and confirmed it to be a bug in 4.3.5 as well.

Resolution is to find this block
```cs
            if (parentNob != null)
            {
                if (parentNob.NetworkBehaviours.Length == 0)
                    Debug.LogError($"{parentNob.gameObject.name} is a parent of {gameObject.name} but it does not contain a NetworkBehaviour. This will cause failure while synchronizing parents. Consider adding {typeof(EmptyNetworkBehaviour).Name} to {parentNob.name} to resolve this problem.");
                else
                    SerializedRootNetworkBehaviour = parentNob.NetworkBehaviours[0];
            }
```

and add this else to it.
```cs
            if (parentNob != null)
            {
                if (parentNob.NetworkBehaviours.Length == 0)
                    Debug.LogError($"{parentNob.gameObject.name} is a parent of {gameObject.name} but it does not contain a NetworkBehaviour. This will cause failure while synchronizing parents. Consider adding {typeof(EmptyNetworkBehaviour).Name} to {parentNob.name} to resolve this problem.");
                else
                    SerializedRootNetworkBehaviour = parentNob.NetworkBehaviours[0];
            }
            else
            {
                SerializedRootNetworkBehaviour = null;
            }
```

Resolved in 4.3.6

**Comment by FREEZX** [comment]
Looks good to me, thanks!
